### PR TITLE
Update GuzzleHttp and adjust tests

### LINKED ISSUE #104: Dependabot alert on use of `guzzlehttp/psr7`
image

Could this be looked into?

### PR DESCRIPTION
**Pull request by wavedeck**
I've upgraded `GuzzleHttp` to the latest minor version and refactored the tests to remove deprecation warnings, caused by GuzzleHttp deprecating `ClientInterface::getConfig` (removing it in the upcoming major version - v8.0)

I've also locked `firebase/php-jwt` to v6.3.x using the tilde (~) selector rather than caret (^) this is, because installing packages using the previous behaviour would install minor versions up to 6.x.x, which would raise the required PHP version to 7.4 (see release notes of [firebase/php-jwt@6.5.0](https://github.com/firebase/php-jwt/releases/tag/v6.5.0))

It's noteworthy, that [php-jwt@6.4.0](https://packagist.org/packages/firebase/php-jwt#v6.4.0) still supports PHP 7.2 just fine and will be the last minor release not requiring to raise the required PHP version for the client.

fixes #104 
closes #101

### CODE REVIEW

File: tests/src/RestTestTrait.php [hunk 1]
```diff
@@ -64,4 +66,8 @@ protected function setCookie($name, $value, $domain): CookieJar {
     return $this->cookies;
   }
 
+  protected function getDomain(): string {
+	return parse_url($this->httpClientConfig['base_uri'], PHP_URL_HOST);
```
Comment (sun) [inline tests/src/RestTestTrait.php:70]:
```suggestion
    return parse_url($this->httpClientConfig['base_uri'], PHP_URL_HOST);
```

File: tests/src/AccessTokenTest.php [hunk 2]
```diff
@@ -3,13 +3,17 @@
...
 
 final class AccessTokenTest extends TestCase {
 
   use RestTestTrait;
 
-  public function testToken(): string {
+	/**
+	 * @throws GuzzleException
+	 */
+	public function testToken(): string {
```
Comment (sun) [inline tests/src/AccessTokenTest.php:16]:
```suggestion
    /**
     * @throws GuzzleException
     */
    public function testToken(): string {
```

File: composer.json [hunk 3]
```diff
@@ -15,12 +15,14 @@
     ],
     "minimum-stability": "stable",
     "require": {
-        "firebase/php-jwt": "^6.3"
+        "firebase/php-jwt": "~6.3.0"
```
Comment (sun) [inline composer.json:18]: Could we move this proposed change into a separate PR, please?  It's highly debatable and will require some discussion that will block the other expected and fine changes of this PR.

After reverting, ensure to run `composer update --lock`

**Review by sun — CHANGES_REQUESTED** [review]
Thanks a lot for creating this PR, @wavedeck! 👌 

I think we could move forward fairly quickly with the Guzzle update.  See my comments below.

File: composer.json (hunk 3 above)
Comment (wavedeck) [reply composer.json]: my latest two commits should do the trick!

**Review by sun — APPROVED** [review]
Thank you! 👍

**Review by dominic-ks — APPROVED** [review]
Thanks, @wavedeck, I've run the tests with the changes in their PR and the updated deps, and they are still passing.
